Slim/Extras/Middleware/FeatherBB.php

```diff
@@ -1,36 +1,132 @@
 
-	public function __construct()
+	public function __construct(array $user_forum_settings = array())
 	{
+        // Define forum env constants
+        $this->data['forum_env'] = array(
+                                    'FEATHER' => true,
+                                    'FEATHER_ROOT' => dirname(__FILE__).'/../../../',
+                                    'FORUM_VERSION' => '1.0.0',
+                                    'FORUM_NAME' => 'FeatherBB',
+                                    'FORUM_DB_REVISION' => 21,
+                                    'FORUM_SI_REVISION' => 2,
+                                    'FORUM_PARSER_REVISION' => 2,
+                                    'FORUM_CACHE_DIR' => dirname(__FILE__).'/../../../cache/', // TODO : Move in user settings
+                                    'FEATHER_UNVERIFIED' => 0,
+                                    'FEATHER_ADMIN' => 1,
+                                    'FEATHER_MOD' => 2,
+                                    'FEATHER_GUEST' => 3,
+                                    'FEATHER_MEMBER' => 4,
+                                    'FEATHER_MAX_POSTSIZE' => 32768,
+                                    'FEATHER_SEARCH_MIN_WORD' => 3,
+                                    'FEATHER_SEARCH_MAX_WORD' => 20,
+                                    'FORUM_MAX_COOKIE_SIZE' => 4048,
+                                    'FEATHER_SHOW_QUERIES' => 1
+                                    );
+
+        // Define forum settings / TODO : handle settings with a class / User input overrides all previous settings
+        $this->data['forum_settings'] = array_merge(self::load_default_settings(), $user_forum_settings);
 
+        // Load DB settings
+        require_once $this->data['forum_env']['FEATHER_ROOT'].'include/idiorm.php';
+        DB::configure('mysql:host='.$this->data['forum_settings']['db_host'].';dbname='.$this->data['forum_settings']['db_name']);
+        DB::configure('username', $this->data['forum_settings']['db_user']);
+        DB::configure('password', $this->data['forum_settings']['db_pass']);
+        DB::configure('logging', true);
+        DB::configure('driver_options', array(\PDO::MYSQL_ATTR_INIT_COMMAND => 'SET NAMES utf8'));
+        DB::configure('id_column_overrides', array(
+            'groups' => 'g_id',
+        ));
 	}
 
-    public function call() 
+    public function load_forum_config()
+    {
+        global $feather_config; // Legacy
+
+        if (file_exists($this->data['forum_env']['FORUM_CACHE_DIR'].'cache_config.php')) {
+            include $this->data['forum_env']['FORUM_CACHE_DIR'].'cache_config.php';
+        } else {
+            require_once $this->data['forum_env']['FEATHER_ROOT'].'include/cache.php';
+            generate_config_cache();
+            require $this->data['forum_env']['FORUM_CACHE_DIR'].'cache_config.php';
+        }
+        return $feather_config;
+    }
+
+    public static function load_default_settings()
```

Comment:
This function is called when the middleware merges the current settings with the forum config stored in the DB, and returns the config array : pointless, but it's a wrapper for the cache loading procedure in order for the call function to be cleaner :)